xdsl/dialects/arith.py

```diff
@@ -190,6 +191,18 @@ class SignlessIntegerBinaryOperation(IRDLOperation, abc.ABC):
 
     assembly_format = "$lhs `,` $rhs attr-dict `:` type($result)"
 
+    @staticmethod
+    def py_operation(lhs: int, rhs: int) -> int | None:
+        return None
+
+    @staticmethod
+    def is_right_zero(attr: AnyIntegerAttr) -> bool:
+        return False
+
+    @staticmethod
+    def is_right_unit(attr: AnyIntegerAttr) -> bool:
+        return False
```

Comment:
Can you add documentation on that function?
Why is there a `right` in the name? It doesn't seem that it takes the left/right side into account?

Reply:
`i` being a right unit means that `x . i = x` but we don't necessarily have `i . x = x`. See the shift operations for example